web3/_utils/module_testing/eth_module.py

```diff
@@ -485,8 +485,8 @@ def test_eth_getBlockByHash(self, web3, empty_block):
         assert block['hash'] == empty_block['hash']
 
     def test_eth_getBlockByHash_not_found(self, web3, empty_block):
-        block = web3.eth.getBlock(UNKNOWN_HASH)
-        assert block is None
+        with pytest.raises(ValueError):
```

Comment:
I realize that I'm late to the party on this one, but I think that this should be a custom exception for each object type.

- `BlockNotFound`
- `TransactionNotFound` (use this for receipt fetching too)
- anything else?

cc @njgheorghita